src/TBuffer.cpp

```diff
@@ -4431,3 +4440,71 @@ void TBuffer::encodingChanged(const QString& newEncoding)
         }
     }
 }
+
+QString TBuffer::processSupportsRequest(const QString& elements)
+{
+    // strip initial SUPPORT and tokenize all of the requested elements
+    auto elementsList = elements.trimmed().remove(0, 7).split(QStringLiteral(" "), QString::SkipEmptyParts);
+    QStringList result;
+
+    auto reportEntireElement = [](auto element, auto& result) {
```

Comment:
I believe you are overdoing the `auto` thing here in the argument list as someone reading the function without a specialized IDE (like here on github) has a hard time figuring out what it means. But that's just food for your thought and nothing I want another review iteration over.